lexer/impl/lexer.cpp

```diff
@@ -1,6 +1,97 @@
 #include "lexer.hpp"
+#include "token.hpp"
+#include <vector>
+#include <string>
+#include "trie.hpp"
+#include <iostream>
+#include <fstream>
+using namespace std;
```

Comment:
We don't use `using namespace std;`. This statement exposes everything from the `std` namespace to the global namespace, which will just pollute it and you won't be able to name your variables that are already defined in `std`, like `max()` or `lower_bound()`.